src/main/java/com/minecolonies/coremod/colony/crafting/CustomRecipe.java

```diff
@@ -391,16 +395,18 @@ public boolean isValidForBuilding(IBuildingWorker building)
         {
             if(Boolean.TRUE.equals(IGlobalResearchTree.getInstance().hasUnlockAbilityEffect(excludedResearchId)))
             {
-                excludedEffect = true;
+                requiredEffect = true;
             }
             else
             {
-                for (ILocalResearch research : colony.getResearchManager().getResearchTree().getCompletedResearch())
+                if(Boolean.TRUE.equals(colony.getResearchManager().getResearchTree().hasCompletedResearch(excludedResearchId)))
                 {
-                    if(research.getId().equals(excludedResearchId) && MinecoloniesAPIProxy.getInstance().getGlobalResearchTree().hasUnlockAbilityEffect(research.getId()))
+                    for(IResearchEffect effect : IGlobalResearchTree.getInstance().getEffectsForResearch(excludedResearchId))
                     {
-                        excludedEffect = true;
-                        //return false;
+                        if(effect instanceof UnlockAbilityResearchEffect)
+                        {
+                            requiredEffect = true;
```

Comment:
This also looks off

Reply:
Yeah, those seem to be it.  Testing against Crusher and Bakery.

Reply:
Takes about 30 seconds to update the recipes being used, working now.  The hut recipe list doesn't seem to update for much longer (I had to open/close the world), not sure if that's worth fixing at this point.

Reply:
Eh, I've not see the UI out of date like that, but if resolution is working with the right counts, I'm good.